### ISSUE TITLE
[Enhancement]: Run initCommands before adding secrets

### ISSUE THREAD

**Issue by sanfelice** [testcontainers/testcontainers-java#5510]
### Module

Vault

### Proposal

Currently I'm trying to enable a new kv engine by using `this.withInitCommand()` method and I'm trying to add secrets to this newly created kv using `this.withSecretInVault()`, but I noticed that in `VaultContainer` class, first the secrets are added and then the init commands are executed. Shouldn't it be the other way round? Because it makes sense for me running init commands to setup the vault the way I want before starting adding secrets to it.

This is the current code:
```java
@Override
protected void containerIsStarted(InspectContainerResponse containerInfo) {
    addSecrets();
    runInitCommands();
}
```
Can't it be like this?
```java
@Override
protected void containerIsStarted(InspectContainerResponse containerInfo) {
    runInitCommands();
    addSecrets();
}
```

**Comment by eddumelendez** [comment]
@sanfelice thanks for raising the issue. As you can see in the following snippet

https://github.com/testcontainers/testcontainers-java/blob/de1324ed2800eff4da326d0c23d281399d006bc0/modules/vault/src/test/java/org/testcontainers/vault/VaultContainerTest.java#L25-L33

line 33 allows to enable a secret engine and then just start using it by passing 2 commands. Take into account that `addSecrets` uses `vault kv put` meanwhile `withSecretInVault` just uses `vault`

**Comment by sanfelice** [comment]
@eddumelendez That's good but would it be a problem running init commands before adding the secrets? 
This is basically what I was trying to achieve
```java
        .withInitCommand("secrets enable -version=2 -path=myApp kv")
        .withSecretInVault(
                "myApp/test",
                "secret_one=password1",
                "secret_two=password2",
)
```
I was wondering if this would be possible, to use the method already available in the class instead of writing everything in the `.withInitCommand` method.

**Comment by eddumelendez** [comment]
TBH, I would rely more on `withInitCommand(String... commands)` because it only pre-append `vault` and `withSecretInVault` is too attached to an specific command offered by vault

```java
.withInitCommand("secrets enable -version=2 -path=myApp kv",
                  "kv put myApp/test",
                  "kv put secret_one=password1",
                  "kv put secret_two=password2"")
```

is it work for you? This way offers more freedom without relying too much on `withSecretInVault` which as you can see can create confusion and I totally think you are right by changing the methods call order but also thinking about the usage of those we should not have that many because one `withSecretInVault` is already covered in the other `withInitCommand ` but just adding a few more words `kv put `

**Comment by sanfelice** [comment]
Yeah, it works. I agree with you that this way we have more freedom to do things the way we want.

Thanks for your help.

**Comment by akhil-ghatiki** [comment]
A slight improvement on @eddumelendez answer. `kv put myApp/test",` this gives an error that body should be present. Below example creates the secrets in a specific path and has to be retrieved using the same path. 

```
.withInitCommand("secrets enable -version=2 -path=myApp kv",
                    "kv put myApp/test/secret_one secret_one=testEncryptionKeyOne",
                    "kv put myApp/test/secret_two secret_two=testEncryptionKeyTwo",
                    "kv put myApp/test/secret_three secret_three=testEncryptionKeyThree");
```
